docs/DeveloperGuide.md

```diff
@@ -302,11 +302,222 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
   Use case ends.
 
-* 3a. The given index is invalid.
+* 3a. The given index is out of range
 
-    * 3a1. AddressBook shows an error message.
+    * 3a1. ContaX shows an error message indicating that the given index is invalid.
 
-      Use case resumes at step 2.
+    * Use case resumes at step 2.
+
+* 3b. The given index is of invalid format
+
+    * 3b1. ContaX shows an error message indicating that the given command is of invalid format.
+
+    * Use case resumes at step 2.
+
+**Use case: Edit a person**
+
+**MSS**
+1. User requests to list persons
+2. ContaX shows a list of persons
+3. User requests to modify a specific person by specifying the corresponding index and parameter flags followed by the new values to change to
+4. ContaX edits the person accordingly
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. The given index is out of range
+
+    * 3a1. ContaX shows an error message indicating that the given index is invalid.
+
+    * Use case resumes at step 2.
+
+* 3b. The given index is of invalid format
+
+    * 3b1. ContaX shows an error message indicating that the given command is of invalid format.
+
+    * Use case resumes at step 2.
+
+* 3c. The phone tag is specified and the given phone is of incorrect format
+   
+    * 3c1. ContaX shows an error message indicating that the given phone number is of invalid format. 
+
+    * Use case resumes at step 2.
+
+*  3d. The email flag is specified and the given email is of incorrect format
+
+    * 3c1. ContaX shows an error message indicating that the given email is of invalid format.
+
+    * Use case resumes at step 2.
+
+*  3e. The tag flag is specified and it is not followed by any tag name
+   
+    * 3e1. ContaX removes all the person's tags.
+
+    * Use case ends.
+
+**Use case: Delete a appointment**
+
+**MSS**
+
+1.  User requests to list appointments
+2.  ContaX shows a list of appointments
+3.  User requests to delete a specific appointments in the list
+4.  ContaX deletes the appointment
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 3a. The given index is out of range
+
+    * 3a1. ContaX shows an error message indicating that the given index is invalid.
+
+    * Use case resumes at step 2.
+
+* 3b. The given index is of invalid format
+
+    * 3b1. ContaX shows an error message indicating that the given command is of invalid format.
+
+    * Use case resumes at step 2.
```

Comment:
Similarly, we can generalize to command syntax error.

Reply:
Fixed